amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/terminal/TerminalManager.java

```diff
@@ -121,17 +124,24 @@ public String executeScript(String terminalId, String catalog, String script) {
           TerminalSessionFactory.SessionConfigOptions.catalogProperty(catalog, key), value);
     }
 
-    TerminalSessionContext context;
     synchronized (sessionMapLock) {
-      sessionMap.computeIfAbsent(
+      sessionMap.compute(
           sessionId,
-          id ->
-              new TerminalSessionContext(
-                  id, metaStore, executionPool, sessionFactory, configuration));
-
-      context = sessionMap.get(sessionId);
+          (id, ctx) -> {
```

Comment:
Changes here seem to fix another issue.

Reply:
Yes, because I found that session configuration was not update after changing server catalog configurations